animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/ClassFileVisitor.java

```diff
@@ -130,26 +128,38 @@ else if ( ( file.getName().endsWith( ".jar" ) || file.getName().endsWith( ".jmod
      *             or a class file (in which case that single class is processed).
      */
     public void process( Path path )
-            throws IOException {
-        Files.walkFileTree(path, Collections.emptySet(), 10000, new SimpleFileVisitor<Path>() {
+        throws IOException
+    {
+        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
 
-            @Override
-            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                return FileVisitResult.CONTINUE;
-            }
+            final SortedSet<Path> files = new TreeSet<>();
 
             @Override
-            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                 if (file.getFileName().toString().endsWith(".class")) {
-                    process(file.toString(), Files.newInputStream(file));
+                    files.add(file);
                 }
                 // XXX we could add processing of jars here as well
                 // but it's not necessary for processing: Paths.get(URI.create("jrt:/modules"))
                 return FileVisitResult.CONTINUE;
             }
 
             @Override
-            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+            public FileVisitResult postVisitDirectory(final Path dir, final IOException exc) throws IOException {
+                if (exc != null) {
+                    throw exc;
+                }
+                for (final Path file : files) {
+                    try (final InputStream inputStream = Files.newInputStream(file)) {
```

Comment:
why final? not sure it helps anything here

Reply:
Not intentional. Not harmful. It does not change the behavior here.

In my company we decided to make anything final, that can be final.
It' supposed to make the semantics of the code clearer when reading code.

Remove the finals if you want.

p.s. thanks for the merge.